src/CentralPackageVersions/Sdk/Sdk.targets

```diff
@@ -191,6 +207,18 @@
       File="$(MSBuildProjectFullPath)" />
   </Target>
 
+  <Target Name="RestoreOriginalPackageReferenceWithVersion"
```

Comment:
This won't work for legacy CSPROJ projects (ones that don't use Microsoft.NET.Sdk) because in Visual Studio it does not run targets before restoring.  This means you must create the list in a top-level `<ItemGroup />` and cannot manipulate anything in a target.

Reply:
Not sure if duplicate declaration will lead into any problem.

- Project declared `PackageReference` with `Version`, but not declared in `Packages.props`
  This should be ok
- Project declared `PackageReference` with `Version`, and also declared in `Packages.props`
  `PackageReference`s in `Packages.props` should override version specified in project file, I guess
- Project declared `PackageReference` with `VersionOverride`, but not declared in `Packages.props`
  This should cause error
- Project declared `PackageReference` with `VersionOverride`, and also declared in `Packages.props`
  This is normal case

Reply:
Yes those cases look correct.  Here's an example of how to do it all in the top-level item groups and avoid a target:
```xml
<Project>
  <!-- Pretend this is declared in the project -->
  <ItemGroup>
    <PackageReference Include="Foo" Version="1.0.0" />
    <PackageReference Include="Bar" />
  </ItemGroup>

  <!-- This is declared in Sdk.targets -->
  <ItemGroup>
    <_PackageReferenceWithVersion Include="@(PackageReference->HasMetadata('Version'))" />
  </ItemGroup>

  <!-- Pretend this is importing Packages.props -->
  <ItemGroup>
    <PackageReference Update="Foo" Version="2.0.0" />
  </ItemGroup>

  <!-- Add this to Sdk.targets to override any versions set by Packages.props and set them back to their original value from the project -->
  <ItemGroup Condition="'$(AllowPackageReferenceWithVersion)' == 'true'">
    <PackageReference Update="@(_PackageReferenceWithVersion)" Version="%(_PackageReferenceWithVersion.Version)" />
  </ItemGroup>

  <Target Name="Build">
    <Message Text="%(PackageReference.Identity) - %(PackageReference.Version)" Importance="High" />
  </Target>
</Project>
```

Reply:
Thanks for the example! Patched.